ui/src/dashboards/apis/v2/index.ts

```diff
@@ -168,3 +169,43 @@ export const updateView = async (
 
   return viewWithIDs
 }
+
+export const addCellUpdateView = async (
+  dashboard: Dashboard,
+  cell: Cell,
+  view: View
+): Promise<View> => {
+  const createdCell = await addCell(dashboard.id, cell)
+  return updateView(dashboard.id, createdCell.id, view)
+}
+
+export const cloneDashboard = async (dashboardToClone: Dashboard) => {
+  const cellsToClone = dashboardToClone.cells
+
+  const dashboardNoCells = _.omit(dashboardToClone, ['cells'])
+
+  const createdDashboard = await createDashboard(dashboardNoCells)
+
+  const cellsLookup = {}
+
+  const viewPromises = []
+
+  cellsToClone.forEach(c => {
+    viewPromises.push(readView(dashboardToClone.id, c.id))
+    cellsLookup[c.id] = c
+  })
+
+  const views = await Promise.all(viewPromises)
+
+  const updatedViews = []
+
+  views.forEach(view => {
```

Comment:
_nit_ It might be a little easier to understand this function if it didn't have these imperative loopy bits. Maybe something like this?

```ts
export const cloneDashboard = async (dashboardToClone: Dashboard) => {
  const dashboardNoCells = _.omit(dashboardToClone, ['cells'])

  const createdDashboard = await createDashboard(dashboardNoCells)

  const cells = dashboardToClone.cells

  const pendingViews = cells.map(c => readView(dashboardToClone.id, c.id))

  const views = await Promise.all(pendingViews)

  const pendingUpdatedViews = views.map(view => {
    const cell = cells.find(c => c.id === view.id)

    return addCellUpdateView(createdDashboard, cell, view)
  })

  await Promise.all(pendingUpdatedViews)

  return createdDashboard
}

```